src/twisted/trial/_dist/test/test_bracket.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Tests for ``twisted.trial._dist.control.bracket``.
+"""
+
+from functools import partial
+from typing import Any, Callable, List
+
+from attrs import define
+from hamcrest import assert_that, equal_to, instance_of, is_
+
+from ....internet.defer import fail, succeed
+from ...runner import TestLoader, TrialSuite
+from ...unittest import SynchronousTestCase
+from ..control import bracket
+
+
+@define
+class BracketTests:
+    """
+    Tests for ``bracket``.
+    """
+
+    case: SynchronousTestCase
+    wrap_success: Callable
+    wrap_failure: Callable
+
+    def test_success(self) -> None:
+        """
+        ``bracket`` calls ``first`` then ``between`` then ``last`` and returns a
+        ``Deferred`` that fires with the result of ``between``.
+        """
+        expected = object()
+        actions: List[str] = []
+        first = partial(actions.append, "first")
+
+        def between() -> Any:
+            actions.append("between")
+            return self.wrap_success(expected)
+
+        last = partial(actions.append, "last")
+        actual = self.case.successResultOf(bracket(first, last, between))
+        assert_that(
+            actual,
+            is_(expected),
+        )
+        assert_that(
+            actions,
+            equal_to(["first", "between", "last"]),
+        )
+
+    def test_failure(self) -> None:
+        """
+        ``bracket`` calls ``first`` then ``between`` then ``last`` and returns a
+        ``Deferred`` that fires with the failure result of ``between``.
+        """
+
+        class SomeException(Exception):
+            pass
+
+        actions: List[str] = []
+        first = partial(actions.append, "first")
+
+        def between() -> Any:
+            actions.append("between")
+            return self.wrap_failure(SomeException())
+
+        last = partial(actions.append, "last")
+        result = self.case.failureResultOf(bracket(first, last, between)).value
+        assert_that(
+            result,
+            instance_of(SomeException),
+        )
+        assert_that(
+            actions,
+            equal_to(["first", "between", "last"]),
+        )
+
+    def test_success_with_failing_last(self) -> None:
```

Comment:
Naming convention: 

```suggestion
    def test_successWithFailingLast(self) -> None:
```

And other tests below.